kernel/aster-nix/src/syscall/dup.rs

```diff
@@ -1,12 +1,38 @@
 // SPDX-License-Identifier: MPL-2.0
 
-use super::{SyscallReturn, SYS_DUP, SYS_DUP2};
+use super::{SyscallReturn, SYS_DUP, SYS_DUP2, SYS_DUP3};
 use crate::{
-    fs::file_table::{FdFlags, FileDesc},
+    fs::file_table::{FdFlags, FileDesc, FileTable},
     log_syscall_entry,
     prelude::*,
+    process::ResourceType,
 };
 
+fn do_dup3(old_fd: FileDesc, new_fd: FileDesc, flags: FdFlags) -> Result<SyscallReturn> {
```

Comment:
In this each source file, put high-level, public methods and types closer to the top of the file and low-level, private methods and types closer to the end of the file. This ordering is good for readability. Here, `do_dup3` should be put after the two public system call functions.

Reply:
Thanks for pointing out that.